src/Microsoft.AspNet.OData.Shared/DefaultODataAPIHandler.cs

```diff
@@ -0,0 +1,157 @@
+    /// and will do these actions. This will be used when the original collection to be Patched is provided.
+    /// </summary>
+    /// <typeparam name="TStructuralType"></typeparam>
+    internal class DefaultODataAPIHandler<TStructuralType> : ODataAPIHandler<TStructuralType> where TStructuralType :class
+    {
+        Type _clrType;
+        ICollection<TStructuralType> originalList;
+
+        public DefaultODataAPIHandler(ICollection<TStructuralType> originalList)
+        {
+            this._clrType = typeof(TStructuralType);
+            this.originalList = originalList?? new List<TStructuralType>();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryGet(IDictionary<string, object> keyValues, out TStructuralType originalObject, out string errorMessage)
+        {
+            ODataAPIResponseStatus status = ODataAPIResponseStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = default(TStructuralType);
+
+            try
+            {
+                originalObject = GetFilteredItem(keyValues);
+
+                if (originalObject == null)
+                {
+                    status = ODataAPIResponseStatus.NotFound;
+                }
+            }
+            catch (Exception ex)
+            {
+                status = ODataAPIResponseStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryCreate(IDictionary<string, object> keyValues, out TStructuralType createdObject, out string errorMessage)
+        {
+            createdObject = default(TStructuralType);
+            errorMessage = string.Empty;
+
+            try
+            {
+                if(originalList != null)
+                {
+                    originalList = new List<TStructuralType>();
+                }
+
+                createdObject = Activator.CreateInstance(_clrType) as TStructuralType;
+                originalList.Add(createdObject);
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                TStructuralType originalObject = GetFilteredItem(keyValues);
+                originalList.Remove(originalObject);
```

Comment:
How come unlike for `DefaultEdmODataAPIHandler` there's no check to verify that `originalObject` is non-null?